project-base/app/web/imageResizer.php

```diff
@@ -66,11 +78,45 @@
 
 function getImageFromUrl(string $url): void
 {
-    $image = file_get_contents($url);
+    $ch = curl_init($url);
+    curl_setopt_array($ch, [
+        CURLOPT_RETURNTRANSFER => true,
+        CURLOPT_FOLLOWLOCATION => true,
+        CURLOPT_TIMEOUT => 5,
+        CURLOPT_CONNECTTIMEOUT => 2,
+        CURLOPT_MAXREDIRS => 5,
+        CURLOPT_USERAGENT => 'ImageProxy/1.0',
+        CURLOPT_HTTPHEADER => [
+            'Accept: ' . $_SERVER['HTTP_ACCEPT'],
+        ]
+    ]);
+    $image = curl_exec($ch);
+    $contentType = curl_getinfo($ch, CURLINFO_CONTENT_TYPE);
+    $statusCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+    curl_close($ch);
+
+    if ($statusCode !== 200 || !$image) {
+        header("HTTP/1.0 404 Not Found");
+        exit;
+    }
 
-    header('Content-type: image/' . getExtension($url));
+    $etag = '"' . md5($image) . '"';
+
+    header('X-Robots-Tag: noindex, nofollow');
+
+    // Support 304 Not Modified
+    if (isset($_SERVER['HTTP_IF_NONE_MATCH']) && $_SERVER['HTTP_IF_NONE_MATCH'] === $etag) {
+        header("HTTP/1.1 304 Not Modified");
+        exit;
+    }
+
+    header('Content-Type: ' . $contentType);
     header('Content-Length: ' . strlen($image));
     header('Content-Disposition: inline');
+    header('Cache-Control: public, max-age=604800');
+    header("Last-Modified: " . gmdate('D, d M Y H:i:s') . ' GMT');
```

Comment:
I think the last modified header should be omitted or set to the actual file modified time. 
Setting the current date defeats the purpose of caching.